receiver/awsecscontainermetricsreceiver/receiver_test.go

```diff
@@ -17,28 +17,38 @@ package awsecscontainermetricsreceiver
 import (
 	"context"
 	"errors"
+	"fmt"
 	"io/ioutil"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/consumer/consumertest"
 	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/ecsutil"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsecscontainermetricsreceiver/internal/awsecscontainermetrics"
 )
 
 type fakeRestClient struct {
 }
 
-func (f fakeRestClient) EndpointResponse() ([]byte, []byte, error) {
-	taskStats, err := ioutil.ReadFile("testdata/task_stats.json")
-	if err != nil {
-		return nil, nil, err
+func (f fakeRestClient) GetResponse(path string) ([]byte, error) {
+	var content []byte
```

Comment:
Can I suggest you change it to the following to remove the need for error handling?

```suggestion
func (f fakeRestClient) GetResponse(uri string) ([]byte, error) {
    var testfile string
    switch uri {
    case escutil.TaskMetadataPath:
       testfile = "testdata/task_metadata.json"
    case awscontainermetrics.TaskStatsPath:
       testfile = "testdata/task_stats.json"
    }
    return io.ReadFile(testfile)
}
```

Reply:
thanks, I've updated with the new helper